### PR TITLE
[WIP] bullet.rb lacking pybullet support, assistance requested

### PR DESCRIPTION
**Pull request by ahundt**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

The bullet.rb script does not enable a lot of functionality built into the latest version of bullet, and I would like to update it accordingly. However, I'm running into some configuration difficulty when trying to update with python support which I'd appreciate some assistance with best practices, and I also want to ensure I don't break the requirements of others.

There also appears to be upstream CMake bugs as described in https://github.com/bulletphysics/bullet3/issues/1107.

### CODE REVIEW

File: Formula/bullet.rb [hunk 1]
```diff
@@ -23,15 +23,44 @@ class Bullet < Formula
   deprecated_option "double-precision" => "with-double-precision"
 
   depends_on "cmake" => :build
+  depends_on "enet" => :recommended
+  depends_on :python => :recommended
+  depends_on "numpy" => :recommended
 
   def install
+    dylib = OS.mac? ? "dylib" : "so"
```
Comment (fxcoudert) [inline Formula/bullet.rb]: No `OS.mac?` in core (here and below)

Comment (ahundt) [inline Formula/bullet.rb]: I guess those lines should be added to the linuxbrew fork only then? Fair enough I can make that change

**Comment by ahundt** [comment]
Python is now configured correctly via the upstream cmake changes bulletphysics/bullet3#1107, but I'm not sure if that can be incorporated until they do a release of bullet itself.

**Comment by ilovezfs** [comment]
@ahundt have you requested a new release?

**Comment by ahundt** [comment]
Asked at https://github.com/bulletphysics/bullet3/issues/1264

**Comment by fxcoudert** [comment]
Going to close this for now, waiting for a new release of bullet and pybullet support.

**Comment by ahundt** [comment]
Can this be reopoened? 2.87 is released and the pybullet cmake build should work correctly

**Pull request closed without merging**